### PR TITLE
Added changelog.md to template

### LINKED ISSUE #277: Add CHANGELOG to /template
This CHANGELOG will be used in context of developing terraform module
- [ ] link to official doc for CHANGELOG syntax
- [ ] link to changelog enforcer. but not implementation

https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/.github/workflows/on_pr_master.yml#L57-L67

### PR DESCRIPTION
**Pull request by claesgill**
Closes #277 

Added `CHANGELOG.md` to template folder

Should we also add a section for how to use the CHANGELOG in the README?

### CODE REVIEW

**Comment by zhenik** [comment]
We use [this](https://keepachangelog.com/en/1.0.0/) syntax for CHANGELOG. It could be useful to have that link

**Comment by claesgill** [comment]
> We use [this](https://keepachangelog.com/en/1.0.0/) syntax for CHANGELOG. It could be useful to have that link

But where should we put the link?

**Comment by zhenik** [comment]
- `vagrant-hashistack` in [CONTRIBUTING.md](https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/docs/CONTRIBUTING.md)
- `vagrant-hashistack-template` in [README.md](https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/template/README.md)

**Review by Neha-Sinha2305 — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 ### Added
 - Directories for `example` and `template_example/example`
 - `nomad_acl.tf` from `template_example/example/` added to `example/vagrant_box_example/`
+- Added CHANGELOG.md in template
```
Comment (zhenik) [inline CHANGELOG.md:24]:
```suggestion
- Added CHANGELOG.md in template #277 
```

**Review by zhenik — CHANGES_REQUESTED** [review]

**Review by zhenik — COMMENTED** [review]
Can you make rebase with master? It should fix linter problem

File: template/README.md [hunk 2]
```diff
@@ -340,6 +341,9 @@ terraform apply
 > :bulb: `default` is the name of running VM. You could also use VM `id`.
 To get vm `id` check `vagrant global-status`.
 
+### Module versioning
+Remember to update the changelog and label the PR as a release (major/minor/patch). The CHANGELOG.md should be in this [syntax](https://keepachangelog.com/en/1.0.0/).
```
Comment (pdmthorsrud) [inline template/README.md:345]: The template workflow does actually not have a "release" workflow, so there is no reason to add a major/minor/patch tag. 
```suggestion
### Changelog
The CHANGELOG.md should follow this [syntax](https://keepachangelog.com/en/1.0.0/).
```

File: template/README.md [hunk 3]
```diff
@@ -49,6 +49,7 @@
       1. [Pushing Resources To MinIO With Ansible (Docker image)](#pushing-resources-to-minio-with-ansible-docker-image)
       2. [Fetching Resources From MinIO With Nomad (Docker image)](#fetching-resources-from-minio-with-nomad-docker-image)
    3. [Iteration of the Development Process](#iteration-of-the-development-process)
+   4. [Module versioning](#module-versioning)
```
Comment (pdmthorsrud) [inline template/README.md:52]:
```suggestion
   4. [Changelog](#changelog)
```

**Review by zhenik — APPROVED** [review]

**Pull request merged**